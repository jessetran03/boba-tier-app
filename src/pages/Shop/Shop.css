.shop-tea-ranking ol {
  list-style-position: inside;
}

.shop-tea-ranking li {
  padding: 15px 10px;
  margin: 10px auto;
  border-style: solid;
  border-radius: 25px;
  max-width: 300px;
}

.shop-tea-list ul {
  list-style: none;
  padding: 10px;
}

.shop-tea-list li {
  max-width: 300px;
  margin: 10px auto;
}

.shop-tea-list div {
  border-style: solid;
}