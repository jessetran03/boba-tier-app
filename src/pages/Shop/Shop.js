import React, { Component } from 'react'
import Comments from '../../components/Comments/Comments'
import ListDrink from '../../components/ListDrink/ListDrink'
import Rating from '../../components/Rating/Rating'
import './Shop.css'
import config from '../../config'
import TokenService from '../../services/token-service'
import STORE from '../../STORE'

export default class Shop extends Component {
  state = {
    listDrinks: STORE.drinks,
    rankedDrinks: STORE.rankedDrinks,
    drinks: [],
    shop: STORE.shop,
    comments: [],
  }
  static defaultProps = {
    match: { params: {} },
  }

  componentDidMount() {
    this.getShop()
    this.getRankedDrinks()
    TokenService.hasAuthToken()
      ? this.getUserDrinks()
      : this.getDrinks()
  }

  getShop = () => {
    const { shopId } = this.props.match.params
    return fetch(`${config.API_ENDPOINT}/shops/${shopId}`, {
      method: 'GET',
    })
      .then(res =>
        (!res.ok)
          ? res.json().then(e => Promise.reject(e))
          : res.json()
      )
      .then((shop) => {
        this.setState({ shop })
      })
      .catch(error => {
        console.error({ error })
      })
  }

  getRankedDrinks = () => {
    const { shopId } = this.props.match.params
    return fetch(`${config.API_ENDPOINT}/ratings/shops/${shopId}`, {
      method: 'GET',
    })
      .then(res =>
        (!res.ok)
          ? res.json().then(e => Promise.reject(e))
          : res.json()
      )
      .then((rankedDrinks) => {
        this.setState({ rankedDrinks })
      })
      .catch(error => {
        console.error({ error })
      })
  }

  getUserDrinks = () => {
    const { shopId } = this.props.match.params
    return fetch(`${config.API_ENDPOINT}/drinks/shops/${shopId}`, {
      method: 'GET',
      headers: {
        'authorization': `bearer ${TokenService.getAuthToken()}`
      }
    })
      .then(res =>
        (!res.ok)
          ? res.json().then(e => Promise.reject(e))
          : res.json()
      )
      .then((drinks) => {
        this.setState({ drinks })
      })
      .catch(error => {
        console.error({ error })
      })
  }

  getDrinks = () => {
    return fetch(`${config.API_ENDPOINT}/drinks`, {
      method: 'GET',
    })
      .then(res =>
        (!res.ok)
          ? res.json().then(e => Promise.reject(e))
          : res.json()
      )
      .then((drinks) => {
        this.setState({ drinks })
      })
      .catch(error => {
        console.error({ error })
      })
  }

  handleAddDrink = e => {
    e.preventDefault()
    const { shopId } = this.props.match.params
    const newDrink = {
      drink_name: e.target['drink-name'].value,
      shop_id: shopId,
    }
    fetch(`${config.API_ENDPOINT}/drinks`, {
      method: 'POST',
      headers: {
        'content-type': 'application/json',
        'authorization': `bearer ${TokenService.getAuthToken()}`
      },
      body: JSON.stringify(newDrink),
    })
      .then(res => {
        if (!res.ok)
          return res.json().then(e => Promise.reject(e))
        return res.json()
      })
      .then(drink => {
        this.getDrinks()
        e.target['drink-name'].value = ''
      })
      .catch(error => {
        console.error({ error })
      })
  }

  render() {
    const rankedDrinks = this.state.rankedDrinks
    const drinks = this.state.drinks
    const shop = this.state.shop
    const shopId = parseInt(this.props.match.params.shopId)
    return (
      <>
        <h2>{shop.shop_name}</h2>
        <h4><i>{shop.city}</i></h4>

        <section className="shop-tea-ranking">
          <h3>Top 5 Drinks</h3>
          {rankedDrinks.length === 0 &&
            <p><i>Looks like there are no ratings for this store yet. Leave some ratings!</i></p>
          }
          <ol>
            {rankedDrinks
              .slice(0, 5)
              .map(drink =>
                <li key={drink.id}>
                  {drink.drink_name}
                  <br /><br />
                  <Rating rating={Math.round(drink.average_rating)} />
                  <br />
                  ({drink.rating_count} Ratings)
                  <br />
                  Rating: {parseFloat(drink.average_rating).toFixed(2)}
                </li>
              )
            }
          </ol>
        </section>

        <section className="shop-tea-list">
          <h3>List of drinks:</h3>
          {TokenService.hasAuthToken() 
            ? <h4>Leave your rating!</h4>
            : <p>Please log in to leave a rating</p>}
          {drinks.length === 0 &&
            <p><i>Looks like no drinks have been added to this store yet. Add some drinks below!</i></p>
          }
          <ul>
            {drinks
              .filter(drink => drink.shop_id === shopId)
              .sort((a, b) => a.id > b.id ? 1 : -1)
              .map(drink =>
                <li key={drink.id}>
                  <ListDrink
                    id={drink.id}
                    drink={drink.drink_name}
                    ratingId={drink.rating_id}
                    rating={drink.rating}
                    onGetUserDrinks={this.getUserDrinks}
                  />
                </li>
              )}
          </ul>
          <h5><i>Don't see a drink? Add it below!</i></h5>
          <form onSubmit={this.handleAddDrink}>
            <input
              type='text'
              id='drink-name-input'
              name='drink-name'
              required
            />
            <button type='submit'>
              Add Drink!
            </button>
          </form>
        </section>

        <Comments shopId={shopId} />
      </>
    )
  }
}