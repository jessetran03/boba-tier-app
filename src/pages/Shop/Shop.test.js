import React from 'react'
import ReactDOM from 'react-dom'
import { BrowserRouter } from 'react-router-dom'
import Shop from './Shop'

it('renders Shop without crashing', () => {
  const div = document.createElement('div')
  ReactDOM.render(
    <BrowserRouter>
      <Shop />
    </BrowserRouter>,
    div
  )
  ReactDOM.unmountComponentAtNode(div)
})