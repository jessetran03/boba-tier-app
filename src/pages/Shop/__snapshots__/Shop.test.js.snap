// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`render Shop without data 1`] = `
<section
  className="shop-page"
>
  <h2>
    -
  </h2>
  <h4>
    <i>
      Houston
      , 
      TX
    </i>
  </h4>
  <section
    className="shop-tea-ranking"
  >
    <h3>
      Top 5 Drinks
    </h3>
    <ol>
      <li>
        <span>
           
        </span>
        <p
          className="average-rating"
        >
          Average Rating: 
          0.00
        </p>
        <div
          className="shop-drinks-rating"
        >
          <svg
            aria-hidden="true"
            className="svg-inline--fa fa-star fa-w-18 star-unrated"
            data-icon="star"
            data-prefix="fas"
            focusable="false"
            role="img"
            style={Object {}}
            viewBox="0 0 576 512"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M259.3 17.8L194 150.2 47.9 171.5c-26.2 3.8-36.7 36.1-17.7 54.6l105.7 103-25 145.5c-4.5 26.3 23.2 46 46.4 33.7L288 439.6l130.7 68.7c23.2 12.2 50.9-7.4 46.4-33.7l-25-145.5 105.7-103c19-18.5 8.5-50.8-17.7-54.6L382 150.2 316.7 17.8c-11.7-23.6-45.6-23.9-57.4 0z"
              fill="currentColor"
              style={Object {}}
            />
          </svg>
          <svg
            aria-hidden="true"
            className="svg-inline--fa fa-star fa-w-18 star-unrated"
            data-icon="star"
            data-prefix="fas"
            focusable="false"
            role="img"
            style={Object {}}
            viewBox="0 0 576 512"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M259.3 17.8L194 150.2 47.9 171.5c-26.2 3.8-36.7 36.1-17.7 54.6l105.7 103-25 145.5c-4.5 26.3 23.2 46 46.4 33.7L288 439.6l130.7 68.7c23.2 12.2 50.9-7.4 46.4-33.7l-25-145.5 105.7-103c19-18.5 8.5-50.8-17.7-54.6L382 150.2 316.7 17.8c-11.7-23.6-45.6-23.9-57.4 0z"
              fill="currentColor"
              style={Object {}}
            />
          </svg>
          <svg
            aria-hidden="true"
            className="svg-inline--fa fa-star fa-w-18 star-unrated"
            data-icon="star"
            data-prefix="fas"
            focusable="false"
            role="img"
            style={Object {}}
            viewBox="0 0 576 512"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M259.3 17.8L194 150.2 47.9 171.5c-26.2 3.8-36.7 36.1-17.7 54.6l105.7 103-25 145.5c-4.5 26.3 23.2 46 46.4 33.7L288 439.6l130.7 68.7c23.2 12.2 50.9-7.4 46.4-33.7l-25-145.5 105.7-103c19-18.5 8.5-50.8-17.7-54.6L382 150.2 316.7 17.8c-11.7-23.6-45.6-23.9-57.4 0z"
              fill="currentColor"
              style={Object {}}
            />
          </svg>
          <svg
            aria-hidden="true"
            className="svg-inline--fa fa-star fa-w-18 star-unrated"
            data-icon="star"
            data-prefix="fas"
            focusable="false"
            role="img"
            style={Object {}}
            viewBox="0 0 576 512"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M259.3 17.8L194 150.2 47.9 171.5c-26.2 3.8-36.7 36.1-17.7 54.6l105.7 103-25 145.5c-4.5 26.3 23.2 46 46.4 33.7L288 439.6l130.7 68.7c23.2 12.2 50.9-7.4 46.4-33.7l-25-145.5 105.7-103c19-18.5 8.5-50.8-17.7-54.6L382 150.2 316.7 17.8c-11.7-23.6-45.6-23.9-57.4 0z"
              fill="currentColor"
              style={Object {}}
            />
          </svg>
          <svg
            aria-hidden="true"
            className="svg-inline--fa fa-star fa-w-18 star-unrated"
            data-icon="star"
            data-prefix="fas"
            focusable="false"
            role="img"
            style={Object {}}
            viewBox="0 0 576 512"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M259.3 17.8L194 150.2 47.9 171.5c-26.2 3.8-36.7 36.1-17.7 54.6l105.7 103-25 145.5c-4.5 26.3 23.2 46 46.4 33.7L288 439.6l130.7 68.7c23.2 12.2 50.9-7.4 46.4-33.7l-25-145.5 105.7-103c19-18.5 8.5-50.8-17.7-54.6L382 150.2 316.7 17.8c-11.7-23.6-45.6-23.9-57.4 0z"
              fill="currentColor"
              style={Object {}}
            />
          </svg>
        </div>
        <p
          className="rating-count"
        >
          (
           
           Ratings)
        </p>
      </li>
      <li>
        <span>
           
        </span>
        <p
          className="average-rating"
        >
          Average Rating: 
          0.00
        </p>
        <div
          className="shop-drinks-rating"
        >
          <svg
            aria-hidden="true"
            className="svg-inline--fa fa-star fa-w-18 star-unrated"
            data-icon="star"
            data-prefix="fas"
            focusable="false"
            role="img"
            style={Object {}}
            viewBox="0 0 576 512"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M259.3 17.8L194 150.2 47.9 171.5c-26.2 3.8-36.7 36.1-17.7 54.6l105.7 103-25 145.5c-4.5 26.3 23.2 46 46.4 33.7L288 439.6l130.7 68.7c23.2 12.2 50.9-7.4 46.4-33.7l-25-145.5 105.7-103c19-18.5 8.5-50.8-17.7-54.6L382 150.2 316.7 17.8c-11.7-23.6-45.6-23.9-57.4 0z"
              fill="currentColor"
              style={Object {}}
            />
          </svg>
          <svg
            aria-hidden="true"
            className="svg-inline--fa fa-star fa-w-18 star-unrated"
            data-icon="star"
            data-prefix="fas"
            focusable="false"
            role="img"
            style={Object {}}
            viewBox="0 0 576 512"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M259.3 17.8L194 150.2 47.9 171.5c-26.2 3.8-36.7 36.1-17.7 54.6l105.7 103-25 145.5c-4.5 26.3 23.2 46 46.4 33.7L288 439.6l130.7 68.7c23.2 12.2 50.9-7.4 46.4-33.7l-25-145.5 105.7-103c19-18.5 8.5-50.8-17.7-54.6L382 150.2 316.7 17.8c-11.7-23.6-45.6-23.9-57.4 0z"
              fill="currentColor"
              style={Object {}}
            />
          </svg>
          <svg
            aria-hidden="true"
            className="svg-inline--fa fa-star fa-w-18 star-unrated"
            data-icon="star"
            data-prefix="fas"
            focusable="false"
            role="img"
            style={Object {}}
            viewBox="0 0 576 512"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M259.3 17.8L194 150.2 47.9 171.5c-26.2 3.8-36.7 36.1-17.7 54.6l105.7 103-25 145.5c-4.5 26.3 23.2 46 46.4 33.7L288 439.6l130.7 68.7c23.2 12.2 50.9-7.4 46.4-33.7l-25-145.5 105.7-103c19-18.5 8.5-50.8-17.7-54.6L382 150.2 316.7 17.8c-11.7-23.6-45.6-23.9-57.4 0z"
              fill="currentColor"
              style={Object {}}
            />
          </svg>
          <svg
            aria-hidden="true"
            className="svg-inline--fa fa-star fa-w-18 star-unrated"
            data-icon="star"
            data-prefix="fas"
            focusable="false"
            role="img"
            style={Object {}}
            viewBox="0 0 576 512"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M259.3 17.8L194 150.2 47.9 171.5c-26.2 3.8-36.7 36.1-17.7 54.6l105.7 103-25 145.5c-4.5 26.3 23.2 46 46.4 33.7L288 439.6l130.7 68.7c23.2 12.2 50.9-7.4 46.4-33.7l-25-145.5 105.7-103c19-18.5 8.5-50.8-17.7-54.6L382 150.2 316.7 17.8c-11.7-23.6-45.6-23.9-57.4 0z"
              fill="currentColor"
              style={Object {}}
            />
          </svg>
          <svg
            aria-hidden="true"
            className="svg-inline--fa fa-star fa-w-18 star-unrated"
            data-icon="star"
            data-prefix="fas"
            focusable="false"
            role="img"
            style={Object {}}
            viewBox="0 0 576 512"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M259.3 17.8L194 150.2 47.9 171.5c-26.2 3.8-36.7 36.1-17.7 54.6l105.7 103-25 145.5c-4.5 26.3 23.2 46 46.4 33.7L288 439.6l130.7 68.7c23.2 12.2 50.9-7.4 46.4-33.7l-25-145.5 105.7-103c19-18.5 8.5-50.8-17.7-54.6L382 150.2 316.7 17.8c-11.7-23.6-45.6-23.9-57.4 0z"
              fill="currentColor"
              style={Object {}}
            />
          </svg>
        </div>
        <p
          className="rating-count"
        >
          (
           
           Ratings)
        </p>
      </li>
      <li>
        <span>
           
        </span>
        <p
          className="average-rating"
        >
          Average Rating: 
          0.00
        </p>
        <div
          className="shop-drinks-rating"
        >
          <svg
            aria-hidden="true"
            className="svg-inline--fa fa-star fa-w-18 star-unrated"
            data-icon="star"
            data-prefix="fas"
            focusable="false"
            role="img"
            style={Object {}}
            viewBox="0 0 576 512"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M259.3 17.8L194 150.2 47.9 171.5c-26.2 3.8-36.7 36.1-17.7 54.6l105.7 103-25 145.5c-4.5 26.3 23.2 46 46.4 33.7L288 439.6l130.7 68.7c23.2 12.2 50.9-7.4 46.4-33.7l-25-145.5 105.7-103c19-18.5 8.5-50.8-17.7-54.6L382 150.2 316.7 17.8c-11.7-23.6-45.6-23.9-57.4 0z"
              fill="currentColor"
              style={Object {}}
            />
          </svg>
          <svg
            aria-hidden="true"
            className="svg-inline--fa fa-star fa-w-18 star-unrated"
            data-icon="star"
            data-prefix="fas"
            focusable="false"
            role="img"
            style={Object {}}
            viewBox="0 0 576 512"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M259.3 17.8L194 150.2 47.9 171.5c-26.2 3.8-36.7 36.1-17.7 54.6l105.7 103-25 145.5c-4.5 26.3 23.2 46 46.4 33.7L288 439.6l130.7 68.7c23.2 12.2 50.9-7.4 46.4-33.7l-25-145.5 105.7-103c19-18.5 8.5-50.8-17.7-54.6L382 150.2 316.7 17.8c-11.7-23.6-45.6-23.9-57.4 0z"
              fill="currentColor"
              style={Object {}}
            />
          </svg>
          <svg
            aria-hidden="true"
            className="svg-inline--fa fa-star fa-w-18 star-unrated"
            data-icon="star"
            data-prefix="fas"
            focusable="false"
            role="img"
            style={Object {}}
            viewBox="0 0 576 512"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M259.3 17.8L194 150.2 47.9 171.5c-26.2 3.8-36.7 36.1-17.7 54.6l105.7 103-25 145.5c-4.5 26.3 23.2 46 46.4 33.7L288 439.6l130.7 68.7c23.2 12.2 50.9-7.4 46.4-33.7l-25-145.5 105.7-103c19-18.5 8.5-50.8-17.7-54.6L382 150.2 316.7 17.8c-11.7-23.6-45.6-23.9-57.4 0z"
              fill="currentColor"
              style={Object {}}
            />
          </svg>
          <svg
            aria-hidden="true"
            className="svg-inline--fa fa-star fa-w-18 star-unrated"
            data-icon="star"
            data-prefix="fas"
            focusable="false"
            role="img"
            style={Object {}}
            viewBox="0 0 576 512"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M259.3 17.8L194 150.2 47.9 171.5c-26.2 3.8-36.7 36.1-17.7 54.6l105.7 103-25 145.5c-4.5 26.3 23.2 46 46.4 33.7L288 439.6l130.7 68.7c23.2 12.2 50.9-7.4 46.4-33.7l-25-145.5 105.7-103c19-18.5 8.5-50.8-17.7-54.6L382 150.2 316.7 17.8c-11.7-23.6-45.6-23.9-57.4 0z"
              fill="currentColor"
              style={Object {}}
            />
          </svg>
          <svg
            aria-hidden="true"
            className="svg-inline--fa fa-star fa-w-18 star-unrated"
            data-icon="star"
            data-prefix="fas"
            focusable="false"
            role="img"
            style={Object {}}
            viewBox="0 0 576 512"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M259.3 17.8L194 150.2 47.9 171.5c-26.2 3.8-36.7 36.1-17.7 54.6l105.7 103-25 145.5c-4.5 26.3 23.2 46 46.4 33.7L288 439.6l130.7 68.7c23.2 12.2 50.9-7.4 46.4-33.7l-25-145.5 105.7-103c19-18.5 8.5-50.8-17.7-54.6L382 150.2 316.7 17.8c-11.7-23.6-45.6-23.9-57.4 0z"
              fill="currentColor"
              style={Object {}}
            />
          </svg>
        </div>
        <p
          className="rating-count"
        >
          (
           
           Ratings)
        </p>
      </li>
      <li>
        <span>
           
        </span>
        <p
          className="average-rating"
        >
          Average Rating: 
          0.00
        </p>
        <div
          className="shop-drinks-rating"
        >
          <svg
            aria-hidden="true"
            className="svg-inline--fa fa-star fa-w-18 star-unrated"
            data-icon="star"
            data-prefix="fas"
            focusable="false"
            role="img"
            style={Object {}}
            viewBox="0 0 576 512"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M259.3 17.8L194 150.2 47.9 171.5c-26.2 3.8-36.7 36.1-17.7 54.6l105.7 103-25 145.5c-4.5 26.3 23.2 46 46.4 33.7L288 439.6l130.7 68.7c23.2 12.2 50.9-7.4 46.4-33.7l-25-145.5 105.7-103c19-18.5 8.5-50.8-17.7-54.6L382 150.2 316.7 17.8c-11.7-23.6-45.6-23.9-57.4 0z"
              fill="currentColor"
              style={Object {}}
            />
          </svg>
          <svg
            aria-hidden="true"
            className="svg-inline--fa fa-star fa-w-18 star-unrated"
            data-icon="star"
            data-prefix="fas"
            focusable="false"
            role="img"
            style={Object {}}
            viewBox="0 0 576 512"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M259.3 17.8L194 150.2 47.9 171.5c-26.2 3.8-36.7 36.1-17.7 54.6l105.7 103-25 145.5c-4.5 26.3 23.2 46 46.4 33.7L288 439.6l130.7 68.7c23.2 12.2 50.9-7.4 46.4-33.7l-25-145.5 105.7-103c19-18.5 8.5-50.8-17.7-54.6L382 150.2 316.7 17.8c-11.7-23.6-45.6-23.9-57.4 0z"
              fill="currentColor"
              style={Object {}}
            />
          </svg>
          <svg
            aria-hidden="true"
            className="svg-inline--fa fa-star fa-w-18 star-unrated"
            data-icon="star"
            data-prefix="fas"
            focusable="false"
            role="img"
            style={Object {}}
            viewBox="0 0 576 512"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M259.3 17.8L194 150.2 47.9 171.5c-26.2 3.8-36.7 36.1-17.7 54.6l105.7 103-25 145.5c-4.5 26.3 23.2 46 46.4 33.7L288 439.6l130.7 68.7c23.2 12.2 50.9-7.4 46.4-33.7l-25-145.5 105.7-103c19-18.5 8.5-50.8-17.7-54.6L382 150.2 316.7 17.8c-11.7-23.6-45.6-23.9-57.4 0z"
              fill="currentColor"
              style={Object {}}
            />
          </svg>
          <svg
            aria-hidden="true"
            className="svg-inline--fa fa-star fa-w-18 star-unrated"
            data-icon="star"
            data-prefix="fas"
            focusable="false"
            role="img"
            style={Object {}}
            viewBox="0 0 576 512"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M259.3 17.8L194 150.2 47.9 171.5c-26.2 3.8-36.7 36.1-17.7 54.6l105.7 103-25 145.5c-4.5 26.3 23.2 46 46.4 33.7L288 439.6l130.7 68.7c23.2 12.2 50.9-7.4 46.4-33.7l-25-145.5 105.7-103c19-18.5 8.5-50.8-17.7-54.6L382 150.2 316.7 17.8c-11.7-23.6-45.6-23.9-57.4 0z"
              fill="currentColor"
              style={Object {}}
            />
          </svg>
          <svg
            aria-hidden="true"
            className="svg-inline--fa fa-star fa-w-18 star-unrated"
            data-icon="star"
            data-prefix="fas"
            focusable="false"
            role="img"
            style={Object {}}
            viewBox="0 0 576 512"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M259.3 17.8L194 150.2 47.9 171.5c-26.2 3.8-36.7 36.1-17.7 54.6l105.7 103-25 145.5c-4.5 26.3 23.2 46 46.4 33.7L288 439.6l130.7 68.7c23.2 12.2 50.9-7.4 46.4-33.7l-25-145.5 105.7-103c19-18.5 8.5-50.8-17.7-54.6L382 150.2 316.7 17.8c-11.7-23.6-45.6-23.9-57.4 0z"
              fill="currentColor"
              style={Object {}}
            />
          </svg>
        </div>
        <p
          className="rating-count"
        >
          (
           
           Ratings)
        </p>
      </li>
      <li>
        <span>
           
        </span>
        <p
          className="average-rating"
        >
          Average Rating: 
          0.00
        </p>
        <div
          className="shop-drinks-rating"
        >
          <svg
            aria-hidden="true"
            className="svg-inline--fa fa-star fa-w-18 star-unrated"
            data-icon="star"
            data-prefix="fas"
            focusable="false"
            role="img"
            style={Object {}}
            viewBox="0 0 576 512"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M259.3 17.8L194 150.2 47.9 171.5c-26.2 3.8-36.7 36.1-17.7 54.6l105.7 103-25 145.5c-4.5 26.3 23.2 46 46.4 33.7L288 439.6l130.7 68.7c23.2 12.2 50.9-7.4 46.4-33.7l-25-145.5 105.7-103c19-18.5 8.5-50.8-17.7-54.6L382 150.2 316.7 17.8c-11.7-23.6-45.6-23.9-57.4 0z"
              fill="currentColor"
              style={Object {}}
            />
          </svg>
          <svg
            aria-hidden="true"
            className="svg-inline--fa fa-star fa-w-18 star-unrated"
            data-icon="star"
            data-prefix="fas"
            focusable="false"
            role="img"
            style={Object {}}
            viewBox="0 0 576 512"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M259.3 17.8L194 150.2 47.9 171.5c-26.2 3.8-36.7 36.1-17.7 54.6l105.7 103-25 145.5c-4.5 26.3 23.2 46 46.4 33.7L288 439.6l130.7 68.7c23.2 12.2 50.9-7.4 46.4-33.7l-25-145.5 105.7-103c19-18.5 8.5-50.8-17.7-54.6L382 150.2 316.7 17.8c-11.7-23.6-45.6-23.9-57.4 0z"
              fill="currentColor"
              style={Object {}}
            />
          </svg>
          <svg
            aria-hidden="true"
            className="svg-inline--fa fa-star fa-w-18 star-unrated"
            data-icon="star"
            data-prefix="fas"
            focusable="false"
            role="img"
            style={Object {}}
            viewBox="0 0 576 512"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M259.3 17.8L194 150.2 47.9 171.5c-26.2 3.8-36.7 36.1-17.7 54.6l105.7 103-25 145.5c-4.5 26.3 23.2 46 46.4 33.7L288 439.6l130.7 68.7c23.2 12.2 50.9-7.4 46.4-33.7l-25-145.5 105.7-103c19-18.5 8.5-50.8-17.7-54.6L382 150.2 316.7 17.8c-11.7-23.6-45.6-23.9-57.4 0z"
              fill="currentColor"
              style={Object {}}
            />
          </svg>
          <svg
            aria-hidden="true"
            className="svg-inline--fa fa-star fa-w-18 star-unrated"
            data-icon="star"
            data-prefix="fas"
            focusable="false"
            role="img"
            style={Object {}}
            viewBox="0 0 576 512"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M259.3 17.8L194 150.2 47.9 171.5c-26.2 3.8-36.7 36.1-17.7 54.6l105.7 103-25 145.5c-4.5 26.3 23.2 46 46.4 33.7L288 439.6l130.7 68.7c23.2 12.2 50.9-7.4 46.4-33.7l-25-145.5 105.7-103c19-18.5 8.5-50.8-17.7-54.6L382 150.2 316.7 17.8c-11.7-23.6-45.6-23.9-57.4 0z"
              fill="currentColor"
              style={Object {}}
            />
          </svg>
          <svg
            aria-hidden="true"
            className="svg-inline--fa fa-star fa-w-18 star-unrated"
            data-icon="star"
            data-prefix="fas"
            focusable="false"
            role="img"
            style={Object {}}
            viewBox="0 0 576 512"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M259.3 17.8L194 150.2 47.9 171.5c-26.2 3.8-36.7 36.1-17.7 54.6l105.7 103-25 145.5c-4.5 26.3 23.2 46 46.4 33.7L288 439.6l130.7 68.7c23.2 12.2 50.9-7.4 46.4-33.7l-25-145.5 105.7-103c19-18.5 8.5-50.8-17.7-54.6L382 150.2 316.7 17.8c-11.7-23.6-45.6-23.9-57.4 0z"
              fill="currentColor"
              style={Object {}}
            />
          </svg>
        </div>
        <p
          className="rating-count"
        >
          (
           
           Ratings)
        </p>
      </li>
    </ol>
  </section>
  <section
    className="shop-tea-list"
  >
    <h3>
      List of Drinks
    </h3>
    <p
      className="list-notice"
    >
      Please log in to leave a rating
    </p>
    <p>
      <i>
        Looks like no drinks have been added to this store yet. Add some drinks below!
      </i>
    </p>
    <ul />
    <form
      className="add-drink"
      onSubmit={[Function]}
    >
      <h5>
        <i>
          Don't see a drink? Add it below!
        </i>
      </h5>
      <input
        id="drink-name-input"
        name="drink-name"
        placeholder="Name of drink"
        required={true}
        type="text"
      />
      <button
        type="submit"
      >
        Add Drink!
      </button>
    </form>
  </section>
  <section
    className="shop-comments"
  >
    <h3>
      Comments
    </h3>
    <p>
      <i>
        Be the first to comment!
      </i>
    </p>
    <ul
      className="shop-comment-list"
    />
    <form
      onSubmit={[Function]}
    >
      <textarea
        id="text"
        maxLength="500"
        name="text"
        placeholder="Write a comment here"
        required={true}
      />
      <br />
      <button
        type="submit"
      >
        Post Comment
      </button>
    </form>
  </section>
</section>
`;
